### ISSUE TITLE
Unexpected situation when freezing BertForMaskedLM

### ISSUE THREAD

**Issue by jaimeenahn** [huggingface/transformers#8907]
## Environment info

     
- `transformers` version: 3.1.0
- Platform: Linux-4.18.0-147.el8.x86_64-x86_64-with-Ubuntu-16.04-xenial
- Python version: 3.7.7
- PyTorch version (GPU?): 1.6.0 (True)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: Yes and No (It happens in both conditions)
- Using distributed or parallel set-up in script?: Yes and No (It happens in both conditions)

### Who can help
 @LysandreJik 

## Information

Model I am using (Bert, XLNet ...): Bert

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce

1. Load pretrained BertForMaskedLM
```
from transformers import BertForMaskedLM
model = BertForMaskedLM.from_pretrained('bert-base-uncased')
```
2. Check whether gradients in the cls.predictions.decoder layer are calculated
```
print(model.cls.predictions.decoder.weight.requires_grad)
```
Result:
```
True
```
3. Check again after only freezing the bert layer
```
for param in model.bert.parameters():
    param.requires_grad = False

print(model.cls.predictions.decoder.weight.requires_grad)
```
Result:
```
False
```

## Expected behavior

It only happens only on BertForMaskedLM. 

If I tried to freeze only the BertModel, cls.predictions.decoder is also frozen.
But as expected, cls.prediction.transform is not frozen.

The exception only occurs in cls.predictions.decoder .

I don't know it is the way you expected but in my sense, it is a kind of unexpected situation for the ones who try to freeze only the BertModel.

**Comment by LysandreJik** [comment]
I think this is probably so because the `cls.predictions.decoder` is a linear layer, which is tied to the embeddings layer. They're pointing to the same weights, so freezeing one of those would result in freezing the other one.
